golem-service-base/src/service/initial_component_files.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{path::PathBuf, sync::Arc};
+
+use bytes::Bytes;
+use sha2::{Sha256, Digest};
+
+use crate::storage::blob::{BlobStorage, BlobStorageNamespace};
+use golem_common::model::InitialComponentFileKey;
+
+const INITIAL_COMPONENT_FILES_LABEL: &str = "initial_component_files";
+
+/// Service for storing initial component files.
+pub struct InitialComponentFilesService {
+    blob_storage: Arc<dyn BlobStorage + Send + Sync>,
+}
+
+impl InitialComponentFilesService {
+    pub fn new(blob_storage: Arc<dyn BlobStorage + Send + Sync>) -> Self {
+        Self { blob_storage }
+    }
+
+    pub async fn get(
+        &self,
+        key: &InitialComponentFileKey,
+    ) -> Result<Option<Bytes>, String> {
+        self
+            .blob_storage
+            .get_raw(
+                INITIAL_COMPONENT_FILES_LABEL,
+                "get",
+                BlobStorageNamespace::InitialComponentFiles,
+                &PathBuf::from(key.0.clone()),
+            )
+            .await
+    }
+
+    pub async fn put_if_not_exists(
+        &self,
+        bytes: &Bytes,
```

Comment:
In a followup this should be changed to instead stream the data so we can handle files that are larger than memory